packages/ckeditor5-link/src/utils.ts

```diff
@@ -194,6 +196,35 @@ export function openLink( link: string ): void {
 	window.open( link, '_blank', 'noopener' );
 }
 
+/**
+ * Scrolls the view to the desired bookmark or open a link in new window.
+ */
+export function handleLinkOpening( link: string, bookmarkEditing: BookmarkEditing | null ): void {
+	if ( !link.startsWith( '#' ) || !bookmarkEditing ) {
+		openLink( link );
+
+		return;
+	}
+
+	const bookmarkId = link.slice( 1 );
+	const modelBookmark = bookmarkEditing.getElementForBookmarkId( bookmarkId );
+
+	if ( !modelBookmark ) {
+		openLink( link );
+
+		return;
+	}
+
+	bookmarkEditing.editor.model.change( writer => {
+		writer.setSelection( modelBookmark, 'before' );
+	} );
+
+	bookmarkEditing.editor.editing.view.scrollToTheSelection( {
+		alignToTop: true,
+		forceScroll: true
+	} );
```

Comment:
This looks like a separate helper, the rest of the code of this function is a duplicate of `isScrollableToBookmark` helper.